plugins/in_tail/tail_file.c

```diff
@@ -1036,12 +1037,39 @@ static int set_file_position(struct flb_tail_config *ctx,
     if (ctx->db) {
         ret = flb_tail_db_file_set(file, ctx);
         if (ret == 0) {
-            if (file->offset > 0) {
-                ret = lseek(file->fd, file->offset, SEEK_SET);
+            int64_t seek_pos;
+            int has_db_position;
```

Comment:
We need to define variables on the beginning of functions. So, could you move these definitions around L1030?